parser.py

```diff
@@ -1,17 +1,30 @@
 #!/bin/env python
 import argparse
 import pprint
+import re
+
 import grammar
 import visitor
 
+class PreProcessor:
+    @staticmethod
+    def remove_comments(program: str):
```

Comment:
Let's use object oriented design here. In the future we may want to support a variety of preprocessors and having a static method in a class makes this rigid and difficult to change.

Furthermore I'd argue that exposing a method called `remove_comments` kind of defeats the purpose of having a class at all. The preprocessor class should `preprocess` and that's it. The actual operations that happen inside it should not leak outside.